smithy-model/src/main/java/software/amazon/smithy/model/loader/AbstractMutableModelFile.java

```diff
@@ -285,10 +260,106 @@ private <S extends Shape, B extends AbstractShapeBuilder<? extends B, S>> Option
                 builder.addTrait(trait);
             }
             return Optional.of(builder.build());
+        } catch (IllegalStateException e) {
+            if (builder.getShapeType() == ShapeType.MEMBER && ((MemberShape.Builder) builder).getTarget() == null) {
```

Comment:
nit - should we throw a more specific exception for this case?

Reply:
I'm not sure adding any more detail would actually be more clear. We could list all the bound shapes, but that just seems like noise

Reply:
I more meant that you didn’t wouldn’t have to do as much introspection of the exception if there was a more specific type to try/catch. But it’s fine.